content/2. Ondersteunende-informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/Integer/1. Integer.md

```diff
@@ -0,0 +1,52 @@
+---
+title: 1. Integer
+taxonomie:
+  - ib-19.2.Datatype-Primitief-Integer.OI
+  - ib-19.3.Datatype-Primitief-Integer.OI
+---
+
+## Wat is een integer?
+Een integer is een datatype waar een positief of negatief getal in past. Een programmeur kan dit datatype aan een variabel geven en hiermee getallen bewaren, bewerken en uitlezen.
+
+> [!TIP] Casus
+> Stel je hebt een evenement met toegangspoortjes georganiseerd. je wilt graag bijhouden hoeveel mensen het evenement betreden.  
+## Hoe zit een integer in elkaar?
+De integer wordt ingezet door bij het initialiseren `int` voor het variabel-naam te zetten. het variabel krijgt dan het datatype integer.
+
+### Codevoorbeeld
+```C#
+int variabelNaam;
+```
+
+Hier kunnen dan nummers aan toegevoegd worden door het variabel waardes toe te wijzen.
+
+### Codevoorbeeld
+```C#
+int leeftijd;
+leeftijd = 17;
+```
+## Hoe gebruik je een integer?
+Een _integer_ is een gegevenstype dat hele getallen opslaat, zoals 0, -1, 42 of 100. er mogen in een _integer_ dus geen kommagetallen staan. 
+
+De maximale waarde van een integer is 2,147,483,647. Als je over de maximale waarde heengaat door er bijvoorbeeld 1 bij op te tellen, reken je verder vanaf de minimale waarde van een integer(-2,147,483,648). Als je toch een groter getal nodig hebt zijn daar andere datatypes voor, zoals de longint of de [[content/2. Ondersteunende-informatie/3. Processen/4. Implementatieproces/2. Bouwen softwaresysteem/Integer/2. Uint|Uint]].
+
+Er kan met _integers_ gerekend worden door gebruik te maken van de [[3. Wiskundige operatoren|operatoren]].  Omdat een _integer_ geen getallen achter de komma heeft, rond de gedeeld door operator (/) naar beneden af.
+
+
+> [!TIP] Casus
```

Comment:
Misschien wat korter?
Een applicatie kan een integer gebruiken om het aantal aanwezige deelnemers bij een evenement bij te houden. Bij binnenkomst of vertrek wordt deze waarde aangepast om het actuele aantal weer te geven.